6/main.js

```diff
@@ -0,0 +1,65 @@
+function thesmallestnumber(num1, num2, num3, num4) {
+  if (isNaN(Math.min.apply(null, [num1, num2, num3, num4]))) {
+    return "Введіть хоча б 4 числа.";
+  } else {
+    return Math.min.apply(null, [num1, num2, num3, num4]);
+  }
+}
+console.log(thesmallestnumber());
+function greatestcommonden(x = "Введіть хоча б 1 число.", y) {
+  if (x < 0 || y < 0) {
+    return "Одне з введених чисел меньше за 0.";
+  }
+  if (y > x) {
+    return greatestcommonden(y, x);
+  }
+  return !y ? x : greatestcommonden(y, x % y);
+}
+console.log(greatestcommonden());
+function perfectnumber(number) {
```

Comment:
Better naming would be
```suggestion
function isPerfectNumber(number) {
```